sharding/validator_manager_utils.py

```diff
@@ -165,35 +165,48 @@ def call_contract_inconstantly(state, ct, contract_addr, func, args, value=0, st
     return result
 
 
-def call_tx(state, ct, func, args, sender, to, value=0, startgas=STARTGAS, gasprice=GASPRICE):
+def call_tx(state, ct, func, args, sender, to, value=0, startgas=STARTGAS, gasprice=GASPRICE, nonce=None):
     # Transaction(nonce, gasprice, startgas, to, value, data, v=0, r=0, s=0)
     tx = Transaction(
-        state.get_nonce(utils.privtoaddr(sender)), gasprice, startgas, to, value,
+        state.get_nonce(utils.privtoaddr(sender)) if nonce is None else nonce,
+        gasprice, startgas, to, value,
         ct.encode_function_call(func, args)
     ).sign(sender)
     return tx
 
 
-def call_deposit(state, sender_privkey, value, validation_code_addr, return_addr):
+def call_deposit(state, sender_privkey, value, validation_code_addr, return_addr, gasprice=GASPRICE):
     ct = get_valmgr_ct()
     return call_tx(
         state, ct, 'deposit', [validation_code_addr, return_addr],
-        sender_privkey, get_valmgr_addr(), value
+        sender_privkey, get_valmgr_addr(), value, gasprice=gasprice
     )
 
 
-def call_withdraw(state, sender_privkey, value, validator_index, signature):
+def call_withdraw(state, sender_privkey, value, validator_index, signature, gasprice=GASPRICE):
     ct = get_valmgr_ct()
     return call_tx(
         state, ct, 'withdraw', [validator_index, signature],
-        sender_privkey, get_valmgr_addr(), value
+        sender_privkey, get_valmgr_addr(), value, gasprice=gasprice
     )
 
 
-def call_tx_add_header(state, sender_privkey, value, header):
+def get_shard_list(state, valcode_addr):
+    ct = get_valmgr_ct()
+    dummy_addr = b'\xff' * 20
+    shard_list = call_contract_constantly(
+        state, ct, get_valmgr_addr(), 'get_shard_list', [valcode_addr],
+        value=0, startgas=10 ** 20, sender_addr=dummy_addr
+    )
+    return shard_list
+    # assert len(shard_list) == 100
+    # return abi.decode_abi(['bool[100]'], shard_list)[0]
+
+
+def call_tx_add_header(state, sender_privkey, value, header, gasprice=GASPRICE, nonce=None, startgas=300000):
```

Comment:
Why do we add `nonce` as a parameter here? It seems not used yet.

Reply:
Ahh, it was used for handling validator's txqueue, but now I use another way to implement.
So you are right, I don't need to assign `nonce` now. I'll fix it. 👍